spec/unit/webrtc/call.spec.ts

```diff
@@ -926,35 +931,107 @@ describe('Call', function() {
+            } as unknown as RTCPeerConnectionIceEvent;
+
+            beforeEach(async () => {
+                await call.answer();
+                await untilEventSent(
+                    FAKE_ROOM_ID,
+                    EventType.CallAnswer,
+                    expect.objectContaining({}),
+                );
+                mockPeerConn = call.peerConn as unknown as MockRTCPeerConnection;
+            });
 
-            await untilEventSent(
-                FAKE_ROOM_ID,
-                EventType.CallCandidates,
-                expect.objectContaining({
-                    candidates: [
-                        fakeCandidateString,
-                    ],
-                }),
-            );
+            afterEach(() => {
+                jest.useRealTimers();
+            });
+
+            it("sends ICE candidates as separate events if they arrive after the answer", async () => {
+                mockPeerConn!.iceCandidateListener!(fakeCandidateEvent);
+
+                await untilEventSent(
+                    FAKE_ROOM_ID,
+                    EventType.CallCandidates,
+                    expect.objectContaining({
+                        candidates: [
+                            fakeCandidateString,
+                        ],
+                    }),
+                );
+            });
+
+            it("retries sending ICE candidates", async () => {
+                jest.useFakeTimers();
+
+                mockSendEvent.mockRejectedValueOnce(new Error("Fake error"));
+
+                mockPeerConn!.iceCandidateListener!(fakeCandidateEvent);
+
+                await untilEventSent(
+                    FAKE_ROOM_ID,
+                    EventType.CallCandidates,
+                    expect.objectContaining({
+                        candidates: [
+                            fakeCandidateString,
+                        ],
+                    }),
+                );
+
+                mockSendEvent.mockClear();
+
+                await untilEventSent(
+                    FAKE_ROOM_ID,
+                    EventType.CallCandidates,
+                    expect.objectContaining({
+                        candidates: [
+                            fakeCandidateString,
+                        ],
+                    }),
+                );
+            });
+
+            it("gives up on call after 5 attempts at sending ICE candidates", async () => {
+                jest.useFakeTimers();
+
+                mockSendEvent.mockImplementation((roomId: string, eventType: string) => {
+                    if (eventType === EventType.CallCandidates) {
+                        return Promise.reject(new Error());
+                    } else {
+                        return Promise.resolve({ event_id: "foo" });
+                    }
+                });
+
+                mockPeerConn!.iceCandidateListener!(fakeCandidateEvent);
+
+                while (!call.callHasEnded()) {
```

Comment:
If the call never ends will the test fail or will it be stuck in this loop?

Reply:
The test will just time out